jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/SessionContainer.java

```diff
@@ -0,0 +1,127 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http2;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.eclipse.jetty.http2.api.Session;
+import org.eclipse.jetty.io.Connection;
+import org.eclipse.jetty.util.annotation.ManagedObject;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.eclipse.jetty.util.component.Dumpable;
+import org.eclipse.jetty.util.component.DumpableCollection;
+import org.eclipse.jetty.util.component.Graceful;
+import org.eclipse.jetty.util.component.LifeCycle;
+
+@ManagedObject("The container of HTTP/2 sessions")
+public class SessionContainer extends AbstractLifeCycle implements Connection.Listener, Graceful, Dumpable
```

Comment:
Is it really a `Container` in the sense that we use it in Jetty.   Maybe just call it `OpenSessions` ?

Reply:
The previous class was called `AbstractHTTP2ServerConnectionFactory.HTTP2SessionContainer`.
It is really a container in a broader term, so I'd prefer to keep the name.
It is in the HTTP/2 package, so not really that close to `oejuc.Container` class, and we have many other `*Container` classes that don't implement `oejuc.Container`.